amqp/src/main/scala/akka/stream/alpakka/amqp/javadsl/AmqpPublishFlow.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.amqp.javadsl
+
+import java.util.concurrent.CompletionStage
+
+import akka.Done
+import akka.japi.Pair
+import akka.japi.function.Function
+import akka.stream.alpakka.amqp._
+import akka.stream.javadsl.{Flow, Keep}
+import akka.util.ByteString
+
+import scala.compat.java8.FutureConverters._
+
+object AmqpPublishFlow {
+
+  /**
+   * Java API: Creates an [[AmqpPublishFlow]] that accepts (ByteString, passthrough) elements.
+   *
+   * This stage materializes to a CompletionStage<Done>, which can be used to know when the Flow completes, either normally
+   * or because of an amqp failure
+   */
+  def createSimple[O](settings: AmqpSinkSettings): Flow[Pair[ByteString, O], O, CompletionStage[Done]] =
+    Flow
+      .fromFunction(new Function[Pair[ByteString, O], (ByteString, O)] {
```

Comment:
You may use `Flow.of[Pair[ByteString, O]]` and a regular `map` operator.

Reply:
Well I tried it, and remembered why I had done so ^^
Your solution works fine on scala 2.12, but with scala 2.11 I would still need to create this `akka.japi.function.Function` in the map :/
So here, `.fromFunction` is just a more concise version of `.of(classOf[Pair[ByteString, O]]).map(new Function...)`